server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -3245,6 +3242,16 @@ private void doSnapshotShard(SnapshotShardContext context) {
         }
     }
 
+    private static void ensureNotAborted(ShardId shardId, SnapshotId snapshotId, IndexShardSnapshotStatus snapshotStatus, String fileName) {
+        try {
+            snapshotStatus.ensureNotAborted();
+        } catch (Exception e) {
+            logger.debug("[{}] [{}] {} on the file [{}], exiting", shardId, snapshotId, e.getMessage(), fileName);
```

Comment:
Not really sure how useful this debug logging is. I left it in even though it means we have to catch and rethrow the exception. We could just remove it too IMO.